chain-abci/src/app/validate_tx.rs

```diff
@@ -149,7 +146,7 @@ fn execute_enclave_tx(
                 .staking_table
                 .withdraw(trie, state.block_time, address, *amount)
                 .expect("withdraw sanity check");
-            Some(*address)
         }
+        _ => panic!("Trying to execute non-enclave transaction"),
```

Comment:
this "catch-all" is potentially error-prone if a new enclave tx variant is added and it's not handled here

Reply:
Code updated based on review